sympy/solvers/tests/test_ode.py

```diff
@@ -1103,11 +1104,31 @@ def test_solve_ics():
     assert solve_ics([Eq(f(x), C1*sin(x) + C2*cos(x))], [f(x)], [C1, C2],
         {f(0): 1, f(x).diff(x).subs(x, 0): 1}) == {C1: 1, C2: 1}
 
-    # XXX: Ought to be ValueError
-    raises(NotImplementedError, lambda: solve_ics([Eq(f(x), C1*sin(x) + C2*cos(x))], [f(x)], [C1, C2], {f(0): 1, f(pi): 1}))
+    assert solve_ics([Eq(f(x), C1*sin(x) + C2*cos(x))], [f(x)], [C1, C2],
+        {f(0): 1}) == {C2: 1}
+
+    # Some more complex tests
+    K, r, f0 = symbols('K r f0')
+    eq = Eq(f(x).diff(x), r * f(x) * (1 - f(x) / K))
+    assert (dsolve(eq, f(x), ics={f(0): f0})) == Eq(f(x), -K * f0 * exp(r * x) / (
+        (K - f0) * (-f0 * exp(r * x) / (K - f0) - 1)))
```

Comment:
Rather than splitting the line in the middle of an expression or assert statement like this I think it would be better to do:
```python
sol = ...
assert dsolve(...) == sol
```
That way if I can see eq and sol next to each other it's easier to read.